src/org/rascalmpl/uri/file/FileURIResolver.java

```diff
@@ -192,11 +191,12 @@ public static ISourceLocation constructFileURI(String path) {
 	public boolean supportsHost() {
 		return false;
 	}
-	
-	@Override
-	public Charset getCharset(ISourceLocation uri) throws IOException {
-		return null;
-	}
+
+        // Commented out during PR draft; this makes no sense. Is a test dependent on this?
```

Comment:
This should not be commented out.

Some schemes allow us to detect charset, file does not.

Reply:
This is still puzzling me. The default method returns UTF8. should that not happen here too? Why are we returning null here?

Reply:
I'll revert it here. It needs a code comment at the very least.